analyzers/src/SonarAnalyzer.Common/Rules/DeclareTypesInNamespacesBase.cs

```diff
@@ -42,7 +43,7 @@ public abstract class DeclareTypesInNamespacesBase<TSyntaxKind> : SonarDiagnosti
               c =>
               {
                   var declaration = c.Node;
-                  if (c.IsRedundantPositionalRecordContext() || IsInnerTypeOrWithinNamespace(declaration, c.SemanticModel))
+                  if (c.IsRedundantPositionalRecordContext() || IsInnerTypeOrWithinNamespace(declaration, c.SemanticModel) || IsException(c.Node))
```

Comment:
The coverage here cannot be reached because Roslyn no longer invokes the analyzer twice for positional records. The issue has been fixed but we still need the call in order to support Roslyn 4.0.0. 

I think it would be helpful to update the comment from `SyntaxNodeAnalysisContextExtensions.IsRedundantPositionalRecordContext` and mention that we still need this for proper support when running on Roslyn 4.0.0.